From d48f297f2165791829407b6ce7c4d144da4bc1a9 Mon Sep 17 00:00:00 2001
From: mayankkk <mayankkk@google.com>
Date: Tue, 3 Dec 2024 11:33:45 -0800
Subject: [PATCH] Don't allow non-system uids to use "android" as calling
 package.

Bug: 310632322
Test: atest ./BroadcastsSecurityTest/src/android/app/cts/broadcasts/BroadcastRegistrationTest.java
Flag: EXEMPT bugfix
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:f9619f7ab4496bf98b42f00c622c94e60c898cd1)
Merged-In: I87517e9b116e5c6ca4ba05729139455c09a31d53
Change-Id: I87517e9b116e5c6ca4ba05729139455c09a31d53
---
 .../java/com/android/server/am/ActivityManagerService.java     | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/services/core/java/com/android/server/am/ActivityManagerService.java b/services/core/java/com/android/server/am/ActivityManagerService.java
index 7d2ec4cae642..4a4c87dcf686 100644
--- a/services/core/java/com/android/server/am/ActivityManagerService.java
+++ b/services/core/java/com/android/server/am/ActivityManagerService.java
@@ -13811,8 +13811,7 @@ public class ActivityManagerService extends IActivityManager.Stub
                 return null;
             }
             if (callerApp.info.uid != SYSTEM_UID
-                    && !callerApp.getPkgList().containsKey(callerPackage)
-                    && !"android".equals(callerPackage)) {
+                    && !callerApp.getPkgList().containsKey(callerPackage)) {
                 throw new SecurityException("Given caller package " + callerPackage
                         + " is not running in process " + callerApp);
             }
-- 
2.49.0.395.g12beb8f557-goog

